src/components/BaseLayerToggler/BaseLayerToggler.js

```diff
@@ -138,26 +171,61 @@ class BaseLayerToggler extends Component {
     if (!layerService) {
       return;
     }
-
+    layerService.un('change:visible', e => this.updateState(e));
+    layerService.un('change:layers', () => this.resetState());
     this.updateState();
-    layerService.on('change:visible', () => this.updateState());
+    layerService.on('change:visible', e => this.updateState(e));
+    layerService.on('change:layers', () => this.resetState());
   }
 
-  updateState() {
+  updateState(evtLayer) {
+    if (evtLayer && !evtLayer.getIsBaseLayer()) {
+      return;
+    }
+
     const { layerService } = this.props;
+    const { idx } = this.state;
+
     const layers = layerService.getBaseLayers() || [];
-    let idx = layers.findIndex(l => l.getVisible());
-    if (idx === -1 && layers.length > 1) {
-      idx = 0;
+    let newIdx = idx;
+    if (newIdx === -1 && layers.length > 1) {
+      newIdx = 0;
       layers[idx].setVisible(true);
     }
+
     this.setState({
       layers,
-      idx,
+      idx: newIdx,
       layerVisible: layers.length > 1 ? layers[idx] : null,
     });
   }
 
+  resetState() {
+    const { layerService } = this.props;
+    const { layerVisible } = this.state;
+
+    if (!layerVisible) {
+      return;
+    }
+
+    const layers = layerService.getBaseLayers() || [];
+    const idx = layers.indexOf(layerVisible);
+    let newIdx;
+
+    if (idx === -1) {
+      newIdx = 1;
```

Comment:
Why 1 an d not 0 ?

Reply:
Because if the previous visible doesn't exist for the new topic, the first baselayer will be set as visible for the map, and it the next one that needs to be used for the BaseLayerToggler.

Reply:
I believe you on this one ;)